src/box/alter.cc

```diff
@@ -3986,144 +3986,166 @@ on_replace_cluster_clear_id(struct trigger *trigger, void * /* event */)
 	return 0;
 }
 
-/**
- * A trigger invoked on replace in the space _cluster,
- * which contains cluster configuration.
- *
- * This space is modified by JOIN command in IPROTO
- * protocol.
- *
- * The trigger updates the cluster configuration cache
- * with uuid of the newly joined instance.
- *
- * During recovery, it acts the same way, loading identifiers
- * of all instances into the cache. Instance globally unique
- * identifiers are used to keep track of cluster configuration,
- * so that a replica that previously joined a replica set can
- * follow updates, and a replica that belongs to a different
- * replica set can not by mistake join/follow another replica
- * set without first being reset (emptied).
- */
+/** Replica definition. */
+struct replica_def {
+	/** Instance ID. */
+	uint32_t id;
+	/** Instance UUID. */
+	struct tt_uuid uuid;
+};
+
+/** Build replica definition from a _cluster's tuple. */
+static struct replica_def *
+replica_def_new_from_tuple(struct tuple *tuple, struct region *region)
+{
+	uint32_t size = 0;
+	struct replica_def *def = region_alloc_object(
```

Comment:
Done. Is the arena used by these allocs limited? Or can it grow beyond `memtx_memory`? If it can't, it might be a problem, because people sometimes do reach this limit. For example, we have a ticket that deletions don't always work when the limit is reached, because they require some memory. It means the xregion functions might kill the process if people try to do some DDL or a new replica would try to register.

Reply:
This arena is unlimited. (It's runtime arena and it has nothing to do with `memtx_memory`.)